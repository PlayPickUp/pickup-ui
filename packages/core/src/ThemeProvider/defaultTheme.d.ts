import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
