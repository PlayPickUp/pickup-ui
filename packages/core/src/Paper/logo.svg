<svg width="100" height="24" viewBox="0 0 100 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M22.0402 21H25.2322L26.2402 15.48H28.2322C31.1842 15.48 32.7922 14.616 33.6322 12.912C34.1362 11.928 34.4962 10.344 34.4962 9.024C34.4962 6.6 33.0802 5.016 29.7442 5.016H24.9442L22.0402 21ZM29.1922 7.68C30.6322 7.68 31.2562 8.208 31.2562 9.336C31.2562 9.936 31.0642 11.04 30.7762 11.616C30.3682 12.408 29.5762 12.816 28.2562 12.816H26.7202L27.6802 7.68H29.1922ZM34.7747 21H37.9907L40.8947 5.016H37.6787L34.7747 21ZM48.9771 4.68C46.4811 4.68 44.7051 5.712 43.6731 7.8C42.5691 9.984 41.8491 14.496 41.8491 16.536C41.8491 19.872 43.9131 21.36 47.1771 21.36C49.6731 21.36 51.1371 20.376 52.0011 18.672C52.4571 17.808 52.6971 16.728 52.8891 15.576L49.9611 15.24C49.7691 16.128 49.6011 16.848 49.3371 17.328C48.8811 18.216 48.1851 18.576 47.2011 18.576C45.8331 18.576 45.0171 17.856 45.0171 16.344C45.0171 14.52 45.7851 10.464 46.4331 9.096C46.9371 8.016 47.7051 7.464 48.8571 7.464C50.1291 7.464 50.9211 8.064 50.9211 9.312C50.9211 9.792 50.8491 10.392 50.7531 10.896L53.7051 10.752C53.8731 10.008 53.9211 9.288 53.9211 8.784C53.9211 6.192 51.9531 4.68 48.9771 4.68ZM65.7764 21L63.0404 12.072L68.5604 5.016H64.9604L59.1524 12.624L60.5444 5.016H57.4004L54.4964 21H57.6404L58.4324 16.56L60.4004 14.16C61.0004 16.2 62.3924 21 62.3924 21H65.7764ZM68.5447 14.208C68.3287 15.48 68.1847 16.344 68.1847 17.112C68.1847 18.576 68.6647 19.632 69.6247 20.304C70.5607 21 71.7607 21.336 73.2007 21.336C75.5767 21.336 77.3047 20.424 78.2407 18.6C78.7207 17.64 78.9367 16.512 79.3207 14.472L81.0247 5.016H77.9047L76.2247 14.4C75.9607 15.864 75.7447 16.752 75.4567 17.328C75.0727 18.096 74.3527 18.528 73.4167 18.528C72.2887 18.528 71.4007 18 71.4007 16.728C71.4007 16.248 71.5207 15.456 71.6887 14.424L73.3687 5.016H70.2007L68.5447 14.208ZM81.1574 21H84.3494L85.3574 15.48H87.3494C90.3014 15.48 91.9094 14.616 92.7494 12.912C93.2534 11.928 93.6134 10.344 93.6134 9.024C93.6134 6.6 92.1974 5.016 88.8614 5.016H84.0614L81.1574 21ZM88.3094 7.68C89.7494 7.68 90.3734 8.208 90.3734 9.336C90.3734 9.936 90.1814 11.04 89.8934 11.616C89.4854 12.408 88.6934 12.816 87.3734 12.816H85.8374L86.7974 7.68H88.3094Z" fill="white"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M11.7547 3L11.2832 9.96H17.4L8.64531 21L9.11681 14.04H3L11.7547 3Z" fill="white"/>
</svg>
