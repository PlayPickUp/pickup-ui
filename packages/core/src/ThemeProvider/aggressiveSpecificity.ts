/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
