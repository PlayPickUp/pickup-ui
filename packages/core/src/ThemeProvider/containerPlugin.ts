/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
