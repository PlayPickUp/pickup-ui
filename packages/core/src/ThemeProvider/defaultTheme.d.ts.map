{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}