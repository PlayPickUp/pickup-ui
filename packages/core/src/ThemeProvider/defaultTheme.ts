import {
  Breakpoints,
  CreateShadow,
  DefaultGradients,
  DefaultShadows,
  DefaultSpacing,
  DefaultTheme,
  DefaultThemeColors,
  DefaultTypography,
  DefaultZIndex,
  FontFamilies,
  GreenMap,
  GreyMap,
  MediaQuery,
  PurpleMap,
  RedMap,
  ResponsiveFonts,
} from "../types";

// Color me PickUp âš¡
const purple: PurpleMap = {
  light: "#F5F0FF",
  lightBase: "#AFA0FF",
  base: "#7A60FF",
  dark: "#5E24FF",
  transparent: "#EAE5FF",
};

const green: GreenMap = {
  light: "#EAFDF7",
  base: "#25DA95",
};

const grey: GreyMap = {
  light: "#F3F2F5",
  lightBase: "#DDDDDD",
  base: "#9D9D9E",
  dark: "#615E66",
  medium: "#DFDAE6",
};

const red: RedMap = {
  light: "#FFF1F4",
  base: "#FF1244",
};

const colors: DefaultThemeColors = {
  primary: { ...purple },
  secondary: { ...grey },
  green,
  purple,
  grey,
  red,
  black: "#2C2933",
  white: "#FFF",
  success: green.base,
  error: red.base,
};

// Font and Typography
const fontFamilies: FontFamilies = {
  body: "'Inter', Helvetica, sans-serif",
  headline: "'Ringside-Condensed', Helvetica, sans-serif",
};

const fontStyles: ResponsiveFonts = {
  desktop: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 48,
      lineHeight: "54px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 36,
      lineHeight: "40px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "28px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "22px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "16px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 18,
      lineHeight: "30px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 13,
      lineHeight: "20px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
  mobile: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 34,
      lineHeight: "38px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "26px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "15px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 11,
      lineHeight: "13px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 15,
      lineHeight: "24px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 12,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
};

const typography: DefaultTypography = {
  fontFamilies,
  fontStyles,
};

// Fancy ass Material UI inspired shadows
const shadowKeyUmbraOpacity = 0.2;
const shadowKeyPenumbraOpacity = 0.14;
const shadowAmbientShadowOpacity = 0.12;

const createShadow: CreateShadow = (...px) => {
  return [
    `${px[0]}px ${px[1]}px ${px[2]}px ${px[3]}px rgba(0, 0, 0, ${shadowKeyUmbraOpacity})`,
    `${px[4]}px ${px[5]}px ${px[6]}px ${px[7]}px rgba(0, 0, 0, ${shadowKeyPenumbraOpacity})`,
    `${px[8]}px ${px[9]}px ${px[10]}px ${px[11]}px rgba(0, 0, 0, ${shadowAmbientShadowOpacity})`,
  ].join(",");
};

const shadows: DefaultShadows = [
  createShadow(0, 1, 3, 0, 0, 1, 1, 0, 0, 2, 1, -1),
  createShadow(0, 1, 5, 0, 0, 2, 2, 0, 0, 3, 1, -2),
  createShadow(0, 1, 8, 0, 0, 3, 4, 0, 0, 3, 3, -2),
  createShadow(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0),
  createShadow(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0),
  createShadow(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0),
  createShadow(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1),
  createShadow(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2),
  createShadow(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2),
  createShadow(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3),
  createShadow(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3),
  createShadow(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4),
  createShadow(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4),
  createShadow(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4),
  createShadow(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5),
  createShadow(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5),
  createShadow(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5),
  createShadow(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6),
  createShadow(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6),
  createShadow(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7),
  createShadow(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7),
  createShadow(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7),
  createShadow(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8),
  createShadow(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8),
];

// Base Unit Spacings
const spacing: DefaultSpacing = {
  base: 4, // ðŸš¨ changing this value (4) could have VERY BAD CONCEQUENCES! BE VERY SURE, MAYBE YOU JUST NEED TO ADD A VALUE? ðŸ¤”
};

// Default zIndex values
const zIndex: DefaultZIndex = {
  overlay: 999,
};

// Default Border Radius
const borderRadius = 4;

// Media Queries
const mediaQuery: MediaQuery = (minWidth: number) => {
  return `@media (min-width: ${minWidth}px)`;
};

// Breakpoints
const breakpoints: Breakpoints = {
  small: 787,
  medium: 1029,
  large: 1280,
};

// Default Gradient Values
const gradients: DefaultGradients = {
  main: `linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%)`,
  hover: `linear-gradient(112.25deg, #914EFF 35.38%, #5D1BFF 122.56%)`,
  disabled:
    "linear-gradient(112.25deg, rgba(145,78,255,0.2) -20.38%, rgba(93,27,255,0.2) 122.56%)",
};

export const defaultTheme: DefaultTheme = {
  borderRadius,
  breakpoints,
  colors,
  gradients,
  mediaQuery,
  shadows,
  spacing,
  typography,
  zIndex,
};
