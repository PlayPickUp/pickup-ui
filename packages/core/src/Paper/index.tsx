import React from "react";
import { createUseStyles } from "react-jss";
import classNames from "classnames";

import logo from "./logo.svg";

import { DefaultTheme, PaperProps } from "../types";

const logoHeadingHeight = 52;

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    display: "block",
    position: "relative",
    width: "100%",
    height: "auto",
    padding: (props: PaperProps) =>
      props.padding ? props.padding : theme.spacing.base * 4,
    boxShadow: "0px 5px 20px rgba(0, 0, 0, 0.08)",
    borderRadius: theme.spacing.base * 3,
    border: "1px solid #EAEAEB",
    paddingTop: ({ withLogo, padding }) =>
      withLogo
        ? logoHeadingHeight + (padding ? padding : theme.spacing.base * 4)
        : padding
        ? padding
        : theme.spacing.base * 4,
  },
  logoHeading: {
    display: "flex",
    position: "absolute",
    justifyContent: "center",
    alignItems: "center",
    top: 0,
    left: 0,
    width: "100%",
    height: logoHeadingHeight,
    backgroundColor: theme.colors.primary.base,
    textAlign: "center",
    borderTopLeftRadius: theme.spacing.base * 3,
    borderTopRightRadius: theme.spacing.base * 3,
    "& img": {
      display: "inline-block",
      position: "relative",
      width: 100,
      height: "auto",
    },
  },
}));

const Paper: React.FC<PaperProps> = ({
  padding,
  children,
  style,
  className,
  withLogo,
}) => {
  const classes = useStyles({ padding, withLogo });
  return (
    <div className={classNames(classes.root, className)} style={style} data-testid="paper">
      {withLogo ? (
        <div className={classes.logoHeading}>
          <div>
            <img src={logo} alt="PickUp" role="presentation" />
          </div>
        </div>
      ) : null}
      {children}
    </div>
  );
};

export default Paper;
