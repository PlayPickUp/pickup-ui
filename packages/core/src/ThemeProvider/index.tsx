import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
