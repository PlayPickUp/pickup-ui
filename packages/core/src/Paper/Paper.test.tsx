import React from "react";
import { render } from "@testing-library/react";

import ThemeProvider from "../ThemeProvider";
import Paper from ".";

test("Paper renders without crashing, matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <Paper>Hello, World</Paper>
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});
