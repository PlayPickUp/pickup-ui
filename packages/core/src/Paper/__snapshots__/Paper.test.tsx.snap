// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j0 {
  width: 100%;
  border: 1px solid #EAEAEB;
  height: auto;
  display: block;
  position: relative;
  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.08);
  border-radius: 12px;
}

body #PickUpUI .j1 {
  padding: 16px;
  padding-top: 16px;
}

<div>
  <div
    id="PickUpUI"
  >
    <div
      class="j0 j1"
      data-testid="paper"
    >
      Hello, World
    </div>
  </div>
</div>
`;

exports[`Paper with logo heading renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j0 {
  width: 100%;
  border: 1px solid #EAEAEB;
  height: auto;
  display: block;
  position: relative;
  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.08);
  border-radius: 12px;
}

body #PickUpUI .j1 {
  padding: 16px;
  padding-top: 68px;
}

body #PickUpUI .j2 {
  top: 0;
  left: 0;
  width: 100%;
  height: 52px;
  display: flex;
  position: absolute;
  text-align: center;
  align-items: center;
  justify-content: center;
  background-color: #7A60FF;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
}

<div>
  <div
    id="PickUpUI"
  >
    <div
      class="j0 j1"
      data-testid="paper"
    >
      <div
        class="j2"
      >
        <div>
          <img
            alt="PickUp"
            role="presentation"
            src="test-file-stub"
          />
        </div>
      </div>
      Hello, World
    </div>
  </div>
</div>
`;
